model/project_parser.go

```diff
@@ -1338,6 +1350,12 @@ func evaluateBVTasks(tse *taskSelectorEvaluator, tgse *tagSelectorEvaluator, vse
 		if len(names) == 0 {
 			emptySelectors = append(emptySelectors, pbvt.Name)
 		}
+		for _, tag := range tse.tagEval.unmatchedTagNames {
```

Comment:
I understand what this is doing but it's a little awkward/complex to deduce that the reason it's built in this particular way is so that the tag evaluator can set/pass state back to this specific function, which then has to be manually cleared on L1358. I feel like this field is kind of masquerading as something that could just be a return value by setting temporary state.

If it's not too difficult, can we change the return value of `(*tagSelectorEvaluator).evalSelector` to also include the unmatched tags? This would simplify down the state being maintained in the tag selector evaluator and obviate the need for the additional `ignoreUnmatchedTags` field since the callers could just check for no names + unmatched tags in the result (and error/ignore as appropriate).

Otherwise, if it's difficult, can we at least add a comment to the field about its purpose (and why unmatched tag names are a special case)?